src/widgets/ChatterListWidget.cpp

```diff
@@ -143,41 +143,95 @@ QString formatChattersError(HelixGetChattersError error, const QString &message)
 
 }  // namespace
 
-ChatterListWidget::ChatterListWidget(const TwitchChannel *twitchChannel,
-                                     QWidget *parent)
-    : BaseWindow({}, parent)
+void ChatterListWidget::setupUi()
 {
-    this->setWindowTitle("Chatter List - " + twitchChannel->getName());
-    assert(twitchChannel != nullptr);
+    this->setWindowTitle("Chatter List - " + this->twitchChannel_->getName());
+    assert(this->twitchChannel_ != nullptr);
 
     this->setAttribute(Qt::WA_DeleteOnClose);
 
-    auto *dockVbox = new QVBoxLayout();
-    auto *searchBar = new QLineEdit(this);
+    this->dockVbox_ = new QVBoxLayout(this);
+    this->searchBar_ = new QLineEdit(this);
+
+    this->chattersList_ = new QListWidget(this);
+    this->resultList_ = new QListWidget(this);
+
+    this->loadingLabel_ = new QLabel("Loading...", this);
+    this->searchBar_->setPlaceholderText("Search User...");
+
+    auto listDoubleClick = [this](const QModelIndex &index) {
+        const auto itemText = index.data().toString();
+
+        if (!itemText.isEmpty())
+        {
+            this->userClicked(itemText);
+        }
+    };
+
+    QObject::connect(this->chattersList_, &QListWidget::doubleClicked, this,
+                     listDoubleClick);
+
+    QObject::connect(this->resultList_, &QListWidget::doubleClicked, this,
+                     listDoubleClick);
+
+    HotkeyController::HotkeyMap actions{
+        {"delete",
+         [this](const std::vector<QString> &) -> QString {
+             this->close();
+             return "";
+         }},
+        {"accept", nullptr},
+        {"reject", nullptr},
+        {"scrollPage", nullptr},
+        {"openTab", nullptr},
+        {"search",
+         [this](const std::vector<QString> &) -> QString {
+             this->searchBar_->setFocus();
+             this->searchBar_->selectAll();
+             return "";
+         }},
+    };
+
+    getApp()->getHotkeys()->shortcutsForCategory(HotkeyCategory::PopupWindow,
+                                                 actions, this);
+
+    this->dockVbox_->addWidget(this->searchBar_);
+    this->dockVbox_->addWidget(this->loadingLabel_);
+    this->dockVbox_->addWidget(this->chattersList_);
+    this->dockVbox_->addWidget(this->resultList_);
+    this->resultList_->hide();
 
-    auto *chattersList = new QListWidget();
-    auto *resultList = new QListWidget();
+    this->setStyleSheet(this->theme->splits.input.styleSheet);
+    this->setLayout(this->dockVbox_);
 
-    auto *loadingLabel = new QLabel("Loading...");
-    searchBar->setPlaceholderText("Search User...");
+    this->setMinimumWidth(300);
+}
+
+void ChatterListWidget::clearUi()
+{
+    std::cout << "clearing the connected users UI" << std::endl;
```

Comment:
Don't use iostreams, use Qt's [`qCDebug`](https://doc.qt.io/qt-6/qloggingcategory.html#qCDebug) and friends.